manifests/dashboard.pp

```diff
@@ -4,66 +4,108 @@
-                                    },
-                                  ]
+  $indexer_server_host = "https://${indexer_server_ip}:${indexer_server_port}",
+  $dashboard_wazuh_api_credentials = [
+    {
+      'id'       => 'default',
+      'url'      => 'https://localhost',
+      'port'     => '55000',
+      'user'     => 'wazuh-wui',
+      'password' => 'wazuh-wui',
+    },
+  ],
 ) {
 
+  include wazuh::repo
+
+  if $::osfamily == 'Debian' {
+    Class['wazuh::repo'] -> Class['apt::update'] -> Package['wazuh-dashboard']
+  } else {
+    Class['wazuh::repo'] -> Package['wazuh-dashboard']
+  }
+
   # assign version according to the package manager
-  case $::osfamily {
-    'Debian' : {
+  case $facts['os']['family'] {
+    'Debian': {
       $dashboard_version_install = "${dashboard_version}-*"
     }
-    'Linux', 'RedHat' : {
-      $dashboard_version_install = "${dashboard_version}"
+    'Linux', 'RedHat', default: {
+      $dashboard_version_install = $dashboard_version
     }
   }
 
   # install package
-  package { 'Installing Wazuh Dashboard...':
+  package { 'wazuh-dashboard':
     ensure => $dashboard_version_install,
     name   => $dashboard_package,
   }
 
-  include wazuh::certificates
+  require wazuh::certificates
 
-  exec { 'Copy Dashboard Certificates':
+  exec { "ensure full path of ${dashboard_path_certs}":
     path    => '/usr/bin:/bin',
-    command => "mkdir $dashboard_path_certs \
-             && cp /tmp/wazuh-certificates/dashboard.pem  $dashboard_path_certs\
-             && cp /tmp/wazuh-certificates/dashboard-key.pem  $dashboard_path_certs\
-             && cp /tmp/wazuh-certificates/root-ca.pem  $dashboard_path_certs\
-             && chown wazuh-dashboard:wazuh-dashboard -R $dashboard_path_certs\
-             && chmod 500 $dashboard_path_certs\
-             && chmod 400 $dashboard_path_certs/*",
+    command => "mkdir -p ${dashboard_path_certs}",
+    creates => $dashboard_path_certs,
+    require => Package['wazuh-dashboard'],
+  }
+  -> file { $dashboard_path_certs:
+    ensure => directory,
+    owner  => $dashboard_fileuser,
+    group  => $dashboard_filegroup,
+    mode   => '0500',
+  }
 
+  [
+    'dashboard.pem',
+    'dashboard-key.pem',
+    'root-ca.pem',
+  ].each |String $certfile| {
+    file { "${dashboard_path_certs}/${certfile}":
+      ensure  => file,
+      owner   => $dashboard_fileuser,
+      group   => $dashboard_filegroup,
+      mode    => '0400',
+      replace => false,  # only copy content when file not exist
+      source  => "/tmp/wazuh-certificates/${certfile}",
+    }
+  }
+
+  # TODO: Fully manage the opensearch_dashboards.yml and a template file resource
+  file { '/etc/wazuh-dashboard/opensearch_dashboards.yml':
+    owner   => 'wazuh-dashboard',
+    group   => 'wazuh-dashboard',
+    mode    => '0640',
+    content => template('wazuh/opensearch_dashboards_yml.erb'),
+    require => Package[$dashboard_package],
+    notify  => Service[$dashboard_service]
```

Comment:
This is needed else puppet can break when a other value for $dashboard_package is definded.
```suggestion
    require => Package['wazuh-dashboard'],
    notify  => Service['wazuh-dashboard']
```